src/pkg/cli/client/byoc/aws/byoc.go

```diff
@@ -355,9 +372,13 @@ func (b *ByocAws) Delete(ctx context.Context, req *defangv1.DeleteRequest) (*def
 }
 
 // stackDir returns a stack-qualified path, like the Pulumi TS function `stackDir`
-func (b *ByocAws) stackDir(name string) string {
-	ensure(b.PulumiProject != "", "pulumiProject not set")
-	return fmt.Sprintf("/%s/%s/%s/%s", byoc.DefangPrefix, b.PulumiProject, b.PulumiStack, name) // same as shared/common.ts
+func (b *ByocAws) stackDir(name string) (string, error) {
+	projectName, err := b.GetProjectName(context.TODO())
```

Comment:
This to me feels like we should split `client.Client` interface in two parts: a stateless connection part and a "project-aware" part.